clippy_lints/src/field_reassign_with_default.rs

```diff
@@ -0,0 +1,208 @@
+                // binding
+                if stmt_shadows_binding(consequtive_statement, binding_name) {
+                    break;
+                }
+                // find out if and which field was set by this `consequtive_statement`
+                else if let Some((field_ident, assign_rhs)) =
+                    field_reassigned_by_stmt(consequtive_statement, binding_name)
+                {
+                    // extract and store the assigned value for help message
+                    let value_snippet = snippet(cx, assign_rhs.span, "..");
+
+                    // always re-insert set value, this way the latest value is stored for output snippet
+                    assigned_fields.insert(field_ident.name, value_snippet);
+
+                    // also set first instance of error for help message
+                    if first_assign.is_none() {
+                        first_assign = Some(consequtive_statement);
+                    }
+                }
+                // interrupt also if no field was assigned, since we only want to look at consequtive statements
+                else {
+                    break;
+                }
+            }
+
+            // if there are incorrectly assigned fields, do a span_lint_and_note to suggest
+            // construction using `Ty { fields, ..Default::default() }`
+            if !assigned_fields.is_empty() {
+                // take the original assignment as span
+                let stmt = &block.stmts[stmt_idx];
+
+                if let StmtKind::Local(preceding_local) = &stmt.kind {
+                    // if all fields of the struct are not assigned, add `.. Default::default()` to the suggestion.
+                    let ext_with_default = !fields_of_type(&binding_type)
+                        .iter()
+                        .all(|field| assigned_fields.contains_key(&field.name));
+
+                    let field_list = assigned_fields
+                        .into_iter()
+                        .map(|(field, value)| format!("{}: {}", field, value))
+                        .collect::<Vec<String>>()
+                        .join(", ");
+
+                    let sugg = if ext_with_default {
+                        format!("{} {{ {}, ..Default::default() }}", binding_type, field_list)
+                    } else {
+                        format!("{} {{ {} }}", binding_type, field_list)
+                    };
+
+                    // span lint once per statement that binds default
+                    span_lint_and_note(
+                        cx,
+                        FIELD_REASSIGN_WITH_DEFAULT,
+                        first_assign.unwrap_or_else(|| unreachable!()).span,
+                        "field assignment outside of initializer for an instance created with Default::default()",
+                        Some(preceding_local.span),
+                        &format!("consider initializing the variable with `{}`", sugg),
```

Comment:
We could add something like `and remove the relevant assignments` to the suggestion to make it clearer